npm/cli
# tests: parallel tests

Several changes to the test suite to support running tests in parallel,
bringing the overall test time down considerably.

- Replace all literal 1337 and 1234 ports with a custom per-process port
  assignment based on the TAP_CHILD_ID environment variable.
- Add common.pkg as a per-test working directory instead of polluting
  __dirname or accidentally reusing the same working directory for
  multiple tests.
- Rework test config handling so that tests don't rely on config setup
  being run in a particular order.
- Remove the npm-registry-couchapp tests, since it (a) relies on
  CouchDB, (b) is no longer a reliable indicator of registry
  compatibility, and (c) is already superceded in most cases by tests
  that use npm-registry-mock.  (A test suite that runs against a
  reference implementation is a thing that should exist, but not here.)
- Remove the fake-registry logging when TAP_CHILD_ID is set, since this
  is extremely hard to make sense of when running multiple tests in
  parallel.

When Node v6 compatibility is dropped in npm v7, we can upgrade to the
latest version of tap for a bit more speed, dropping Domains (and the
associated deprecation warnings), and a fancier test reporter.

(The readable-stream update fixes a crash on Node v12.  It is separate from the main purpose of this PR, but it was annoying to keep dealing with, and probably a good idea anyway.)
